File: XeeSDK/Model/XeeDevice.m
//
//  XeeDevice.m
//  XeeSDK
//
//  Created by Jean-Baptiste Dujardin on 26/04/2017.
//  Copyright © 2017 Eliocity. All rights reserved.
//

#import "XeeDevice.h"

@implementation XeeDevice

-(instancetype)initWithJSON:(NSDictionary *)JSON {
    self = [super initWithJSON:JSON];
    if (self) {
        if ([JSON isKindOfClass:[NSDictionary class]]) {
            _xeeId = [JSON objectForKey:@"xeeId"];
            _status = [JSON objectForKey:@"status"];
            _associationAttempts = [JSON objectForKey:@"associationAttempts"];
        }
    }
    return self;
}

-(NSString *)description {
    return [NSString stringWithFormat:@"\
            xeeId: %@\r\
            status: %@\r\
            associationAttempts: %@",
            _xeeId, _status, _associationAttempts];
    return nil;
}

@end

File: XeeSDK/Classes/Model/XeeDevice.swift
//
//  XeeDevice.swift
//  XeeSDK
//
//  Created by Jean-Baptiste Dujardin on 04/10/2017.
//

import ObjectMapper

open class XeeDevice: XeeObject {
    
    public var deviceID: String?
    public var brand: String?
    public var vehicleID: String?
    
    public override init() {
        super.init()
    }
    
    required public init?(map: Map) {
        super.init(map: map)
    }
    
    open override func mapping(map: Map) {
        super.mapping(map: map)
        deviceID <- map["id"]
        brand <- map["brand"]
        vehicleID <- map["vehicleId"]
    }
    
}
